examples/pipelines_braindecode/Braindecode_ShallowFBCSPNET.yml

```diff
@@ -0,0 +1,73 @@
+name: Braindecode_ShallowFBCSPNET
+
+paradigms:
+  - LeftRightImagery
+  - MotorImagery
+  - P300
+  - SSVEP
+
+citations:
+  - https://doi.org/10.1002/hbm.23730
+
+pipeline:
+  - name: CreateBraindecodeDataset
+    from: moabb.pipelines.braindecode
+  - name: EEGClassifier
+    from: braindecode
+    parameters:
+      model:
+        - name: ShallowFBCSPNet
+        - from: braindecode.models
+        - parameters:
+            in_chans: ${dataset.in_chans}
+            n_classes: ${dataset.classes}
+            input_windows_size: ${dataset.input_windows_size}
+            final_conv_length: auto
```

Comment:
How to get the value that defines the model during the execution time?

I want to discuss this point to implement our chosen decision, @sylvchev and @carraraig.

Some options so far:

1) We need to get/access this information during the execution, so when we define the pipeline.
- I saw one yaml file with this type of argument that dynamically access the value from the execution environment (I wonder if it will work this way and whether these values will be available when we define the pipeline from the yaml [?]). 

Or, 

2) Adding some dumb values, and we can use the InputSetterEEG that I build in the [Braindecode example](#340) to redefine the model and reinitialize the model when the channel, classes, and input_windows_size change. I think we need to test to see if this addition doesn't change the behaviour somehow.

Or, 

3) build a Lazy Pytorch wrapper... something similar to this [NoisyLazyLinear](https://github.com/sailfish009/rl-1/blob/9f832920ab1e8f639ea1e1d3db5eaf5455162640/torchrl/modules/models/exploration.py#L146), using the [LazyModuleMixin](https://pytorch.org/docs/stable/generated/torch.nn.modules.lazy.LazyModuleMixin.html#torch.nn.modules.lazy.LazyModuleMixin).

Another option that Idk. Which path do I follow?

Reply:
Changing the clone step to _clone dep on the type of pipeline.